projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx

```diff
@@ -182,15 +177,54 @@ export default class DomEventHandler extends Component {
 		}
 	};
 
-	scrollOverlayToTop = event => {
-		// NOTE: the propertyName need to be aligned with the animation
-		if ( event?.propertyName !== 'opacity' ) {
-			return;
+	fixBodyScroll = () => {
+		if ( this.props.isVisible ) {
+			this.preventBodyScroll();
+			// This ensures the search input is visible on mobile devices.
+			// @see https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo
+			window?.scrollTo( 0, 0 );
+		} else if ( ! this.props.isVisible ) {
+			this.restoreBodyScroll();
 		}
-		// @see https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo
-		window?.scrollTo( 0, 0 );
 	};
 
+	/**
+	 * 1) When the overlay is open, we set body to fixed position.
+	 * 2) Body would be scrolled to top, so we need to set top to where the scroll position was.
+	 * 3) And we remember the body postition in `this.top`
+	 */
+	preventBodyScroll() {
+		this.top = parseInt( window.scrollY ) || 0;
```

Comment:
Thanks 😄  Changed in 0583405e2f.
I did consider to put it as a state, but considering it would introduce more complexity then used a variable. But it surely would look more consistent using a state.